File: sample_code/java/Tetris/PlayerWidget.java

import java.awt.BorderLayout;
import java.awt.Dimension;
import java.awt.GridLayout;
import java.awt.Insets;
import javax.swing.JLabel;
import javax.swing.JPanel;
import javax.swing.border.EtchedBorder;
import javax.swing.border.TitledBorder;

public class PlayerWidget extends JPanel {
    private JLabel score;
    private JLabel lines;
    private JLabel level;
    private Engine engine;
    private PlaygroundWidget myPgWidget;
    private NextWidget myNextWidget;

    public NextWidget getNextWidget() {
        return myNextWidget;
    }

    public Engine getEngine() {
        return engine;
    }

    public Insets getInsets() {
        return new Insets(5, 5, 5, 5);
    }

    public void actualizeNumbers(int li, int sc, int le) {
        lines.setText(String.valueOf(li));
        score.setText(String.valueOf(sc));
        level.setText(String.valueOf(le));
    }

    public PlaygroundWidget getPlaygroundWidget() {
        return myPgWidget;
    }

    public PlayerWidget() {
        JPanel panel = new JPanel(), subpanel;

        myPgWidget = new PlaygroundWidget();
        myNextWidget = new NextWidget();
        engine = new Engine(this);
        lines = new JLabel("0");
        score = new JLabel("0");
        level = new JLabel("1");

        setLayout(new BorderLayout());
        panel.setLayout(new GridLayout(4, 1));

        subpanel = new JPanel();
        subpanel.add(myNextWidget);
        subpanel.setBorder(new EtchedBorder());
        panel.add(subpanel);
        subpanel = new JPanel();
        subpanel.add(lines);
        subpanel.setBorder(new TitledBorder("Removed Lines"));
        panel.add(subpanel);
        subpanel = new JPanel();
        subpanel.add(score);
        subpanel.setBorder(new TitledBorder("Score"));
        panel.add(subpanel);
        subpanel = new JPanel();
        subpanel.add(level);
        subpanel.setBorder(new TitledBorder("Level"));
        panel.add(subpanel);
        subpanel = new JPanel();
        subpanel.add(myPgWidget);
        subpanel.setBorder(new EtchedBorder());
        add(subpanel, BorderLayout.WEST);
        add(panel, BorderLayout.EAST);
        setPreferredSize(new Dimension(320, 450));
    }
}

File: sample_code/cplusplus/Tetris/PlayerWidget.cpp

#include "PlayerWidget.h"

Engine * PlayerWidget::getEngine() {
    return engine;
}

NextWidget * PlayerWidget::getNextWidget() {
    return myNextWidget;
}

void PlayerWidget::actualizeNumbers(int li, int sc, int le) {
    lines->display(li);
    score->display(sc);
    level->display(le);
}

PlayerWidget::PlayerWidget() {
    myPgWidget = new PlaygroundWidget();
    myNextWidget = new NextWidget();
    engine = new Engine(this);
    lines = new QLCDNumber(4, 0);
    score = new QLCDNumber(6, 0);
    level = new QLCDNumber(2, 0);
    linesLabel = new QLabel(tr("Lines"));
    scoreLabel = new QLabel(tr("Score"));
    levelLabel = new QLabel(tr("Level"));

    QGridLayout *layout = new QGridLayout;

    layout->addWidget(myPgWidget, 0, 0, 7, 1);
    layout->addWidget(myNextWidget, 0, 1);
    layout->addWidget(linesLabel, 1, 1);
    layout->addWidget(lines, 2, 1);
    layout->addWidget(scoreLabel, 3, 1);
    layout->addWidget(score, 4, 1);
    layout->addWidget(levelLabel, 5, 1);
    layout->addWidget(level, 6, 1);

    setLayout(layout);

    resize(300, 500);
}

PlaygroundWidget * PlayerWidget::getPlaygroundWidget() {
    return myPgWidget;
}
